File: app/src/main/java/com/suncity/dailynotices/islib/ISNav.java
package com.suncity.dailynotices.islib;

import android.app.Activity;
import android.content.Context;
import android.net.Uri;
import android.os.Environment;
import android.support.v4.app.Fragment;
import com.facebook.drawee.view.SimpleDraweeView;
import com.suncity.dailynotices.islib.config.ISCameraConfig;
import com.suncity.dailynotices.islib.config.ISListConfig;
import com.suncity.dailynotices.islib.ui.ISCameraActivity;
import com.suncity.dailynotices.islib.ui.ISListActivity;
import com.suncity.dailynotices.utils.LogUtils;

import java.io.File;

/**
 * 总线
 * <p>
 * Created by yuyuhang on 2017/10/23.
 */
public class ISNav {

    private static ISNav instance;

    public static ISNav getInstance() {
        if (instance == null) {
            synchronized (ISNav.class) {
                if (instance == null) {
                    instance = new ISNav();
                }
            }
        }
        return instance;
    }


    public void displayImage(Context context, String path, SimpleDraweeView imageView) {
        LogUtils.INSTANCE.e("path =" + ("file://" + path));
        imageView.setImageURI(Uri.parse(("file://" + path)));
    }

    public void toListActivity(Object source, ISListConfig config, int reqCode) {
        if (source instanceof Activity) {
            ISListActivity.startForResult((Activity) source, config, reqCode);
        } else if (source instanceof Fragment) {
            ISListActivity.startForResult((Fragment) source, config, reqCode);
        } else if (source instanceof android.app.Fragment) {
            ISListActivity.startForResult((android.app.Fragment) source, config, reqCode);
        }
    }

    public void toCameraActivity(Object source, ISCameraConfig config, int reqCode) {
        if (source instanceof Activity) {
            ISCameraActivity.startForResult((Activity) source, config, reqCode);
        } else if (source instanceof Fragment) {
            ISCameraActivity.startForResult((Fragment) source, config, reqCode);
        } else if (source instanceof android.app.Fragment) {
            ISCameraActivity.startForResult((android.app.Fragment) source, config, reqCode);
        }
    }

}

File: app/src/main/java/com/suncity/dailynotices/islib/ISNav.kt
package com.suncity.dailynotices.islib

import android.app.Activity
import android.content.Context
import android.net.Uri
import androidx.fragment.app.Fragment
import com.facebook.drawee.view.SimpleDraweeView
import com.suncity.dailynotices.islib.config.ISCameraConfig
import com.suncity.dailynotices.islib.config.ISListConfig
import com.suncity.dailynotices.islib.ui.ISCameraActivity
import com.suncity.dailynotices.islib.ui.ISListActivity
import com.suncity.dailynotices.utils.LogUtils


/**
 * 总线
 */
@Suppress("DEPRECATION")
class ISNav {

    /**
     * TODO 需要调试
     */
    fun displayImage(context: Context, path: String, imageView: SimpleDraweeView) {
        imageView.setImageURI(Uri.parse("file://$path"))
    }

    fun toListActivity(source: Any, config: ISListConfig, reqCode: Int) {
        if (source is Activity) {
            ISListActivity.startForResult(source, config, reqCode)
        } else if (source is Fragment) {
            ISListActivity.startForResult(source, config, reqCode)
        }
    }

    fun toCameraActivity(source: Any, config: ISCameraConfig, reqCode: Int) {
        when (source) {
            is Activity -> ISCameraActivity.startForResult(source, config, reqCode)
            is Fragment -> ISCameraActivity.startForResult(source, config, reqCode)
        }
    }

    companion object {

        private var instance: ISNav? = null

        fun getInstance(): ISNav {
            if (instance == null) {
                synchronized(ISNav::class.java) {
                    if (instance == null) {
                        instance = ISNav()
                    }
                }
            }
            return instance!!
        }
    }

}
